cmd/fpga_admissionwebhook/fpga_admissionwebhook.go

```diff
@@ -86,7 +152,98 @@ func getTLSConfig(certFile string, keyFile string) *tls.Config {
 	}
 }
 
-func mutatePods(ar v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
+func getPatchOpsPreprogrammed(containerIdx int, container corev1.Container) ([]string, error) {
+	var ops []string
+
+	for resourceName, resourceQuantity := range container.Resources.Limits {
+		newName := translateFpgaResourceName(resourceName)
+		if len(newName) > 0 {
+			op := fmt.Sprintf(resourceReplaceOp, containerIdx,
+				"limits", rfc6901Escaper.Replace(string(resourceName)),
+				containerIdx, "limits", newName, resourceQuantity.String())
+			ops = append(ops, op)
+		}
+	}
+	for resourceName, resourceQuantity := range container.Resources.Requests {
+		newName := translateFpgaResourceName(resourceName)
+		if len(newName) > 0 {
+			op := fmt.Sprintf(resourceReplaceOp, containerIdx,
+				"requests", rfc6901Escaper.Replace(string(resourceName)),
+				containerIdx, "requests", newName, resourceQuantity.String())
+			ops = append(ops, op)
+		}
+	}
+
+	return ops, nil
+}
+
+func getEnvVars(container corev1.Container) string {
+	var jsonstrings []string
+	for _, envvar := range container.Env {
+		if envvar.Name == "FPGA_REGION" || envvar.Name == "FPGA_AFU" {
+			continue
```

Comment:
I don't think we should skip it silently when variables are already there. It's an error situation from my point of view.

Reply:
+1. If user already set something, it is an error situation. we expect that user only specifies something what admission controller understands. If user tries to trick admission controller - it should be error.

Reply:
now an error is generated